src/features/todo/components/Home/Home.tsx

```diff
@@ -1,17 +1,30 @@
 import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { useQuery } from 'react-query';
+import { useMutation, useQuery } from 'react-query';
 import { AppContainer } from './elements';
 import AddButton from '@/components/Buttons/AddButton';
 import DeleteConfirmDialog from '@/components/Dialogs/DeleteConfirmDialog';
 import EmptyTodoList from '@/components/EmptyTodo/EmptyTodoList';
 import HeaderComponent from '@/components/Header/Header';
 import Notification from '@/components/Notification/Notification';
 import TodoItem from '@/components/Todo/TodoItem/TodoItem';
-import { fetchTodosEndpoint, Todo } from '../../../../api';
+import { Todo, fetchTodosEndpoint, deleteTodoEndpoint } from '@/api';
 
 const Home = () => {
-  const { data: todos } = useQuery<Todo[], Error>('todos', fetchTodosEndpoint);
+  const { data: todos, refetch } = useQuery<Todo[], Error>(
+    'todos',
+    fetchTodosEndpoint,
+  );
+  const mutateDeleteTodo = useMutation(deleteTodoEndpoint, {
+    onSuccess: async () => {
+      await refetch(); // Update data after deletion
```

Comment:
Prefer to use [Query Invalidation](https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation) instead of refetching so that it will be declarative and the tanstack query will handle the refetching

```suggestion
  const queryClient = useQueryClient();
  const { data: todos } = useQuery<Todo[], Error>('todos', fetchTodosEndpoint);

  const mutateDeleteTodo = useMutation(deleteTodoEndpoint, {
    onSuccess: async () => {
        await queryClient.invalidateQueries('todos');
```